src/components/WalletModal/WalletView/index.tsx

```diff
@@ -24,9 +25,16 @@ export default function WalletView({
   const { t } = useTranslation();
 
   const [isLoading, setIsLoading] = useState(false);
+  const [qrCodeUri, setQRCodeUri] = useState<string | null>(null);
 
   const isInstalled = wallet.installed();
 
+  useEffect(() => {
+    if (wallet.connector.onQRCodeURI) {
+      wallet.connector.onQRCodeURI(setQRCodeUri);
+    }
+  }, []);
```

Comment:
do we need to add `wallet` here ?
```suggestion
  }, [wallet]);
```

Reply:
Don't know when change the props it don't force the component to re-render?

Reply:
if you are sure that when component mount at that time `wallet.connector.onQRCodeURI` function exists then we dont need wallet in dependency otherwise we need it.